### PR TITLE
Check that the headers are equal while ignoring the signature for proposer slashings

### PR DESCRIPTION
**Pull request by prestonvanloon**
**What type of PR is this?**

Bug fix

**What does this PR do? Why is it needed?**

We had a minor spec deviation in checking proposer slashings. 

**Which issues(s) does this PR fix?**

Fixes https://github.com/sigp/beacon-fuzz/issues/74

**Other notes for review**

See https://github.com/sigp/beacon-fuzz/issues/74 for more context.

### CODE REVIEW

**Review by rauljordan — APPROVED** [review]
LGTM, thanks @prestonvanloon

File: beacon-chain/core/blocks/proposer_slashing_test.go [hunk 1]
```diff
@@ -181,3 +184,73 @@ func TestProcessProposerSlashings_AppliesCorrectStatus(t *testing.T) {
...
+				},
+				beaconState: beaconState,
+			},
+			wantErr: "expected slashing headers to differ",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			testutil.ResetCache()
+			if err := blocks.VerifyProposerSlashing(tt.args.beaconState, tt.args.slashing); (err != nil || tt.wantErr != "") && err.Error() != tt.wantErr {
```
Comment (nisdas) [inline beacon-chain/core/blocks/proposer_slashing_test.go:251]: we should use our assertion library over here

Comment (prestonvanloon) [reply beacon-chain/core/blocks/proposer_slashing_test.go:251]: Does the assertion library support this behavior? 

I want: "if the test expects an error, assert the error is as expected"

Comment (nisdas) [reply beacon-chain/core/blocks/proposer_slashing_test.go:251]: Not yet, it does have `assert.ErrorContains` which would be the closest to it.  But yeah maybe not worth it to change now.

**Pull request merged**